### DISCUSSION TITLE
Psycopg 2.8.6 on Centos or Ubuntu very slow

### DISCUSSION THREAD

**Discussion by athanaselos** [psycopg/psycopg2#1595] [Q&A]
We run a small Python3 script making a first select to get all the videos whose loop another SELECT reads the information about these videos. This script reads 3 videos per second. The query on the database runs in 2 seconds to read 977880 records.

About 15 days ago, we do about 1,000 reads per second with this same script. Why this performance degradation?

[titi.txt](https://github.com/psycopg/psycopg2/files/11967889/titi.txt)

**Comment by dvarrazzo** [comment]
Most likely it's not a problem with psycopg.

**Comment by athanaselos** [comment]
This came from the extremely fragmented vidéo ID index of over 90.
